src/ArticleVote.jsx

```diff
@@ -0,0 +1,71 @@
+import Button from 'react-bootstrap/Button';
+import { useEffect, useState } from 'react';
+import { applyVote } from './api/article-api';
+import VoteMessage from './VoteMessage';
+
+export default function ArticleVote({ votes, articleID }) {
+    const [vote, setVote] = useState(0);
+    const [voteInfo, setVoteInfo] = useState(null);
+    const [voteAction, setVoteAction] = useState({
+        voted: false,
+    });
+
+    function handleVote(voteInc) {
+        setVote(voteInc)
+        if (voteInc === -1) {
+            setVoteAction((voteAction) => {
+                const newAction = { ...voteAction };
+                newAction.voted = true;
+                newAction.up = false;
+                newAction.down = true;
+                return newAction;
+            })
+        } else {
+            setVoteAction((voteAction) => {
+                const newAction = { ...voteAction };
+                newAction.voted = true;
+                newAction.down = false;
+                newAction.up = true;
+                return newAction;
+            })
+        }
+    }
+
+    useEffect(() => {
+        const patchVote = async () => {
+            if (vote === 1 || vote === -1) {
+                try {
+                    const result = await applyVote(articleID, vote);
+                    setVoteInfo(true)
+                    setTimeout(() => {
+                        setVoteInfo(null);
+                    }, 2000)
+                } catch (err) {
+                    setVoteInfo("Vote failed - try again in a few minutes")
+                    setTimeout(() => {
```

Comment:
Not sure if this setTimout is very user-friendly, it could lead to inconsistencies in the user experience.